src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -7,18 +7,82 @@
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static float threshold;
+
+    private Bucket<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Bucket[DEFAULT_CAPACITY];
+        threshold = DEFAULT_CAPACITY * LOAD_FACTOR;
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            increaseSize();
+        }
+        int bucketIndex = getBucketIndex(key, buckets.length);
+        Bucket<K, V> bucket = getBucket(key, bucketIndex);
+        if (bucket != null) {
+            bucket.value = value;
+        } else {
+            Bucket<K, V> bucketEntry = new Bucket<>(key, value, buckets[bucketIndex]);
+            buckets[bucketIndex] = bucketEntry;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Bucket<K, V> bucket = getBucket(key, getBucketIndex(key, buckets.length));
+        return bucket != null ? bucket.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getBucketIndex(K key, int capacity) {
+        return key != null ? (key.hashCode() >>> 1) % buckets.length : 0;
+    }
+
+    private Bucket getBucket(K key, int index) {
+        Bucket<K, V> bucket = buckets[index];
+        while (bucket != null) {
+            if (key == bucket.key || key != null && key.equals(bucket.key)) {
```

Comment:
since we understand how it works, we can use Objects.equals()

Reply:
thanks, replaced!